citesphere/src/main/java/edu/asu/diging/citesphere/core/zotero/impl/ZoteroConnector.java

```diff
@@ -174,6 +177,52 @@ public Item updateItem(IUser user, Item item, String groupId, List<String> colle
         }
         return getItem(user, groupId, item.getKey());
     }
+    
+    /**
+     * This method makes a call to Zotero to batch update items and return back
+     * response
+     * 
+     * @param groupId      group id of citations
+     * @param items        List of items to be updated
+     * @param ignoreFields Fields that are not necessary to be updated
+     * 
+     * @return ZoteroUpdateItemsResponse returns statuses of items.
+     */
+    @Override
+    public ZoteroUpdateItemsResponse updateItems(IUser user, List<Item> items, String groupId,
+            List<String> ignoreFields) throws ZoteroConnectionException, ZoteroHttpStatusException {
+        Zotero zotero = getApi(user);
+        ItemCreationResponse response = zotero.getGroupsOperations().batchUpdateItems(groupId, items, ignoreFields);
+        try {
+            TimeUnit.SECONDS.sleep(1);
+        } catch (InterruptedException e) {
+            logger.error("Could not sleep.", e);
+        }
+        ZoteroUpdateItemsResponse statuses = new ZoteroUpdateItemsResponse();
+
+        List<String> successItemsKeys = new ArrayList<>();
+        Map<String, String> success = response.getSuccess();
+        for (Map.Entry<String, String> entry : success.entrySet()) {
+            successItemsKeys.add(entry.getValue());
+        }
+        statuses.setSuccessItems(successItemsKeys);
+
+        List<String> unchangedItemsKeys = new ArrayList<>();
+        Map<String, String> unchanged = response.getSuccess();
+        for (Map.Entry<String, String> entry : unchanged.entrySet()) {
+            unchangedItemsKeys.add(entry.getValue());
+        }
+        statuses.setUnchagedItems(unchangedItemsKeys);
+
+        List<String> failedItemsKeys = new ArrayList<>();
+        Map<String, FailedMessage> failed = response.getFailed();
+        for (Map.Entry<String, FailedMessage> entry : failed.entrySet()) {
+            failedItemsKeys.add(entry.getValue().getKey());
+        }
+        statuses.setFailedItems(failedItemsKeys);
```

Comment:
each for each loop, etc can probably replace with a one-liner using the streaming api (using `stream()`, `map()`, and `collect()`).